src/Azure.Functions.ArtifactAssembler/ArtifactAssembler.cs

```diff
@@ -0,0 +1,216 @@
+
+            var packTasks = _customHostTargetArtifactZipNames.Select(async artifactName =>
+            {
+                var runtimeDir = Directory.EnumerateDirectories(_inProc8ExtractedRootDir)
+                                          .FirstOrDefault(dir => dir.Contains(artifactName));
+                if (runtimeDir == null)
+                {
+                    throw new InvalidOperationException($"Runtime directory not found for {artifactName}");
+                }
+
+                var targetRuntimeName = Path.GetFileName(runtimeDir);
+                var targetRuntimeDir = Path.Combine(customHostTargetArtifactDir, targetRuntimeName);
+                Directory.CreateDirectory(targetRuntimeDir);
+
+                // Copy in-proc8 files
+                await Task.Run(() => FileUtilities.CopyDirectory(runtimeDir, Path.Combine(targetRuntimeDir, InProc8DirectoryName)));
+
+                // Copy in-proc6 files if exist
+                var inProc6Dir = Path.Combine(_inProc6ExtractedRootDir, targetRuntimeName);
+                if (Directory.Exists(inProc6Dir))
+                {
+                    await Task.Run(() => FileUtilities.CopyDirectory(inProc6Dir, Path.Combine(targetRuntimeDir, InProc6DirectoryName)));
+                }
+
+                var rid = GetRIdForArtifactName(artifactName);
+
+                Console.WriteLine($"Copying core-tools-host files for {rid}");
+                var coreToolsHostDir = Path.Combine(_coreToolsHostExtractedRootDir, rid);
+                if (Directory.Exists(coreToolsHostDir))
+                {
+                    await Task.Run(() => FileUtilities.CopyDirectory(coreToolsHostDir, targetRuntimeDir));
+                }
+                else
+                {
+                    throw new InvalidOperationException($"Core tools host directory not found for {rid}");
+                }
+
+                // targetRuntimeDir now contains custom core-tools host, in-proc6 and in-proc8 sub directories. Create a zip file.
+                var zipPath = Path.Combine(customHostTargetArtifactDir, $"{targetRuntimeName}.zip");
+                await Task.Run(() => FileUtilities.CreateZipFile(targetRuntimeDir, zipPath));
+                Console.WriteLine($"Created target runtime zip: {zipPath}");
+
+                Directory.Delete(targetRuntimeDir, true);
+            });
+
+            await Task.WhenAll(packTasks);
+
+            // Delete the extracted directories
+            Directory.Delete(_inProc6ExtractedRootDir, true);
+            Directory.Delete(_inProc8ExtractedRootDir, true);
+            Directory.Delete(_coreToolsHostExtractedRootDir, true);
+        }
+
+        private string GetRIdForArtifactName(string artifactName)
+        {
+            if (_artifactNamesToToRIdMap.TryGetValue(artifactName, out var rid))
+            {
+                return rid;
+            }
+
+            throw new InvalidOperationException($"rid not found for artifact name {artifactName}");
```

Comment:
Nit: acronym should be all caps, but honestly we should expand the acronym fully in general.

Reply:
Fixed.